File: BST.java
import java.util.Scanner;

class Node {
    int data;
    Node left, right;

    Node(int x) {
        data = x;
        left = right = null;
    }
}

public class BST {

    static Node insert(Node root, int x) {

        if (root == null) {
            return new Node(x);
        }

        if (x < root.data) {
            root.left = insert(root.left, x);
        } else {
            root.right = insert(root.right, x);
        }

        return root;
    }

    static Node create() {

        Scanner sc = new Scanner(System.in);

        Node root = null;

        System.out.print("Enter node (-1 to stop): ");
        int x = sc.nextInt();

        while (x != -1) {

            root = insert(root, x);

            System.out.print("Enter node (-1 to stop): ");
            x = sc.nextInt();
        }

        return root;
    }

    static void inorder(Node root) {

        if (root != null) {

            inorder(root.left);

            System.out.print(root.data + " ");

            inorder(root.right);
        }
    }

    public static void main(String[] args) {

        Node root = create();

        System.out.print("\nInorder Traversal: ");

        inorder(root);
    }
}

File: BST.cpp
#include<iostream>
using namespace std;

class node
{
    public:
    int data;
    node *left,*right;
    node(int x)
    {
        data=x;
        left=right=NULL;
    }
};

node *insert(node *root, int x)
{
if(root==NULL)
return new node(x);

if(x<root->data)
root->left=insert(root->left,x);
else
root->right=insert(root->right,x);
return root;
}

node *create() {
    int x;
    node *root = NULL;
    cout<<"\nEnter node(-1 to stop): ";
    cin>>x;
    while(x!=-1) {
        root=insert(root,x);
        cout<<"\nEnter node(-1 to stop): ";
        cin>>x;
    }
    return root;
}

void inorder(node * root)
{
    if(root!=NULL)
    {
        inorder(root->left);
        cout<<root->data<<" ";
        inorder(root->right);
    }
}

int main()
{
    node *root=create();
    cout<<"\nInorder Traversal: ";
    inorder(root);
}
